sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -804,15 +807,26 @@ def settle_message_via_receiver_link(
             if settle_operation == MESSAGE_DEFER:
                 return handler.settle_messages(
                     message._delivery_id,
+                    message._delivery_tag,
                     'modified',
                     delivery_failed=True,
                     undeliverable_here=True
                 )
         except AttributeError as ae:
             raise RuntimeError("handler is not initialized and cannot complete the message") from ae
 
+        except AMQPLinkError as le:
+            if (
+                le.condition == ErrorCondition.InternalError
+                and isinstance(le.description, str)
+                and le.description.startswith("Delivery tag")
```

Comment:
Is there any way to avoid checking the description here? It feels mildly unreliable, since pyamqp  will be an external lib in the future. And the description can change (even though it'll be by us), since that's non-breaking. But checking against description here will break us.